src/transformation/visitors/language-extensions/table.ts

```diff
@@ -1,109 +1,44 @@
-    extensions.ExtensionKind.TableAddMethodType,
-];
-
-const tableExtensions = [extensions.ExtensionKind.TableNewType, ...tableCallExtensions];
-
-function getTableExtensionKindForCall(
-    context: TransformationContext,
-    node: ts.CallExpression,
-    validExtensions: extensions.ExtensionKind[]
-) {
-    const type = getFunctionTypeForCall(context, node);
-    return type && validExtensions.find(extensionKind => extensions.isExtensionType(type, extensionKind));
-}
-
-export function isTableExtensionIdentifier(context: TransformationContext, node: ts.Identifier) {
-    const type = context.checker.getTypeAtLocation(node);
-    return tableExtensions.some(extensionKind => extensions.isExtensionType(type, extensionKind));
-}
+import { LanguageExtensionCallTransformer } from "./call-extension";
 
 export function isTableNewCall(context: TransformationContext, node: ts.NewExpression) {
-    const type = context.checker.getTypeAtLocation(node.expression);
-    return extensions.isExtensionType(type, extensions.ExtensionKind.TableNewType);
-}
-
-export function transformTableExtensionCall(
-    context: TransformationContext,
-    node: ts.CallExpression,
-    isOptionalCall: boolean
-): lua.Expression | undefined {
-    const extensionType = getTableExtensionKindForCall(context, node, tableCallExtensions);
-    if (!extensionType) return;
-    if (isOptionalCall) {
-        context.diagnostics.push(unsupportedBuiltinOptionalCall(node));
-        return lua.createNilLiteral();
-    }
-
-    if (
-        extensionType === extensions.ExtensionKind.TableDeleteType ||
-        extensionType === extensions.ExtensionKind.TableDeleteMethodType
-    ) {
-        return transformTableDeleteExpression(context, node, extensionType);
-    }
-
-    if (
-        extensionType === extensions.ExtensionKind.TableGetType ||
-        extensionType === extensions.ExtensionKind.TableGetMethodType
-    ) {
-        return transformTableGetExpression(context, node, extensionType);
-    }
-
-    if (
-        extensionType === extensions.ExtensionKind.TableHasType ||
-        extensionType === extensions.ExtensionKind.TableHasMethodType
-    ) {
-        return transformTableHasExpression(context, node, extensionType);
-    }
-
-    if (
-        extensionType === extensions.ExtensionKind.TableSetType ||
-        extensionType === extensions.ExtensionKind.TableSetMethodType
-    ) {
-        return transformTableSetExpression(context, node, extensionType);
-    }
-
-    if (
-        extensionType === extensions.ExtensionKind.TableAddType ||
-        extensionType === extensions.ExtensionKind.TableAddMethodType
-    ) {
-        return transformTableAddExpression(context, node, extensionType);
-    }
+    return getExtensionKindForNode(context, node.expression) === ExtensionKind.TableNewType;
 }
+export const tableNewExtensions = [ExtensionKind.TableNewType];
+
+export const tableExtensionTransformers: { [P in ExtensionKind]?: LanguageExtensionCallTransformer } = {
```

Comment:
This might also be nicer to locate somewhere in the language-extensions dir and then import

Reply:
Isn't this already in the language-extensions dir?